src/main/resources/pl/edu/agh/iet/dts/ui/static/js/controller/mapController.js

```diff
@@ -35,32 +35,40 @@ app.controller('mapController', ['$http', '$scope', '$rootScope', '$interval', f
         );
     };
 
-    $scope.assignDate = function () {
+    $scope.assignDate = function (timestamp) {
         const format = function (num) {
             return ('0' + num).slice(-2);
         };
 
-        now = new Date();        
+        now = new Date(1000 * timestamp);
         date = now.getFullYear() + '-' + format(now.getMonth() + 1) + '-' + format(now.getDate());
         time = format(now.getHours()) + ':' + format(now.getMinutes());
-        $rootScope.lastSync = date + ' ' + time;
+        $rootScope.lastSync = 'Last synchronization: ' + date + ' ' + time + '.';
+    };
+
+    $scope.havePointsChanged = function (checkpoint, data) {
+        if (checkpoint.length !== data.length) {
+            return true;
+        }
+
+        for (i = 0; i < checkpoint.length; i++) {
+            if (checkpoint[i].latitude !== data[i].latitude) return true;
+            if (checkpoint[i].longitude !== data[i].longitude) return true;
+        }
+
+        return false;
     };
 
     $scope.createRoute = function (data) {
-        if (checkpointArray.length === data.length && checkpointLimit === $rootScope.points) {
+        if (limit === $rootScope.points && !$scope.havePointsChanged(checkpoint, data)) {
             return;
         }
 
-        checkpointLimit = $rootScope.points;
-        checkpointArray = data.slice();
+        checkpoint = data.slice();
+        limit = $rootScope.points;
         let routeRequest;
 
-        num = Math.min($rootScope.points, 10)
-        data = data.slice(Math.max(0, data.length - num));
-
-        if (data.length === 0) {
-            console.log('Nothing to show. There is no GPS data for this user.');
```

Comment:
By the previous comment I meant that this message is not intended to be written into the console, however it should be displayed to a user instead of "Set preferences in order to display a route." when this user has its preferences set and has not any points aggregated on the backend side.